File: Python/Sorting/Selection-Sort.py
def SelectionSort(arr):
    for i in range(len(arr)):
        #Finding minimum element from remaining unsorted array
        min_idx = i  
        for j in range(i+1,len(arr)):
            if arr[min_idx] > arr[j]:
                min_idx = j
        
        #swapping
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

#Taking the input
array = list(map(int,input("Enter your array (as space separated integers) : ").split()))

#Passing array to the function
SelectionSort(array)

print(array)

File: C++/Sorting/Selection-Sort.cpp
#include <iostream>

void Selection_Sort(int arr[], int n);
void swap(int* a, int* b);

int main(){

    int arr[50], n;

    std::cout<< "Enter the size of your array : " ;
    std::cin >> n;

    std::cout<< "Enter the elements of your array : " ;
    for(int i=0; i<n; ++i)
        std::cin >>arr[i];

    Selection_Sort(arr,n);

    std::cout<< "Sorted array : " << std::endl;
    for(int i=0; i<n; ++i)
        std::cout << arr[i] << ",";

    return 0;
}

void Selection_Sort(int arr[], int n){

    for(int i=0; i<n; ++i){
        int min_idx = i;
        for(int  j=i+1; j<n; ++j){
            if (arr[min_idx] > arr[j])
                min_idx = j;
        }
    
        swap(&arr[i], &arr[min_idx]);
    }
}

void swap(int* a, int* b){

    int temp;

    temp = *a;
    *a = *b;
    *b = temp;
}
